src/main/java/sc/fiji/bdvpg/scijava/services/SourceAndConverterBdvDisplayService.java

```diff
@@ -428,56 +429,69 @@ public BdvHandle getB() {
-        sacToBdvHandleRefs.get(sac).add(bhr);
+        sacToBdvRefs.get(sac).add(bhr);
 
         ConverterSetup cs = getConverterSetupsViaReflection(bdvh_in).get(index);
 
         // BigWarp Hack
         if (cs instanceof BigWarpConverterSetupWrapper) {
             BigWarpConverterSetupWrapper wcs = (BigWarpConverterSetupWrapper) cs;
             wcs.getSourceConverterSetup().setViewer(() -> {
-                if ( sacToBdvHandleRefs.get(sac) != null) {
-                    sacToBdvHandleRefs.get(sac).forEach( bhref -> bhref.bdvh.getViewerPanel().requestRepaint());
+                if ( sacToBdvRefs.get(sac) != null) {
+                    sacToBdvRefs.get(sac).forEach( bhref -> bhref.bdv.getViewerPanel().requestRepaint());
                 }
             });
         } else {
             cs.setViewer(() -> {
-                if ( sacToBdvHandleRefs.get(sac) != null) {
-                    sacToBdvHandleRefs.get(sac).forEach( bhref -> bhref.bdvh.getViewerPanel().requestRepaint());
+                if ( sacToBdvRefs.get(sac) != null) {
+                    sacToBdvRefs.get(sac).forEach( bhref -> bhref.bdv.getViewerPanel().requestRepaint());
                 }
             });
         }
-        bss.sourceAndConverterToMetadata.get(sac).put(CONVERTERSETUP, cs);
+        bdvSourceAndConverterService.sacToMetadata.get(sac).put( CONVERTER_SETUP, cs);
     }
 
     /**
      * For debug purposes, check that all is in sync
      */
     public void logLocationsDisplayingSource() {
-        sacToBdvHandleRefs.forEach(( sac, lbdvref) -> {
+        sacToBdvRefs.forEach(( sac, bdvHandleRefs) -> {
             log.accept(sac.getSpimSource().getName()+":"+sac.toString());
-            lbdvref.forEach(bdvref -> {
-                log.accept("\t bdv = "+bdvref.bdvh.toString()+"\t i = "+bdvref.indexInBdv);
+            bdvHandleRefs.forEach(bdvref -> {
+                log.accept("\t bdv = "+bdvref.bdv.toString()+"\t i = "+bdvref.indexInBdv);
             });
         });
     }
 
+    public void updateDisplays(SourceAndConverter[] sacs)
+    {
+        final Set< BdvHandle > bdvHandlesNeedingRepaint = new HashSet<>();
+        for( SourceAndConverter sac : sacs )
+        {
+            if ( sacToBdvRefs.containsKey( sac ) )
+                sacToBdvRefs.get( sac ).forEach( bdvRef -> bdvHandlesNeedingRepaint.add( bdvRef.bdv )  );
+        }
+
+       bdvHandlesNeedingRepaint.forEach( bdv -> bdv.getViewerPanel().requestRepaint() );
+    }
+
     /**
      * Class containing a BdvHandle and an index -> reference to where a Source is located
-     * This could disappear if the indexing logic disappears
+     * TODO: This could disappear if the indexing logic disappears
+     *
      */
     class BdvHandleRef {
-        BdvHandle bdvh;
+        BdvHandle bdv;
         int indexInBdv;
 
-        public BdvHandleRef(BdvHandle bdvh, int idx) {
-            this.bdvh = bdvh;
+        public BdvHandleRef( BdvHandle bdv, int idx) {
+            this.bdv = bdv;
```

Comment:
Why this change ? I thought you liked `bdvHandle` so either `bdvHandle` or `bdvh` but not `bdv` only.

Reply:
I think I asked in an issue whether it would be ok to change `bdvHandle` => `bdv`, and I think you said yes, but we could also stick with `bdvHandle`.

Reply:
ok, maybe I misunderstood, I think it's good to keep bdvHandle then. It's more consistent with what we have everywhere else.